runtime/compiler/env/VMJ9.cpp

```diff
@@ -5043,6 +5043,52 @@ TR_J9VMBase::getMemberNameFieldKnotIndexFromMethodHandleKnotIndex(TR::Compilatio
    return knot->getOrCreateIndex(mnObject);
    }
 
+TR::KnownObjectTable::Index
+TR_J9VMBase::getMethodHandleTableEntryIndex(TR::Compilation *comp, TR::KnownObjectTable::Index vhIndex, TR::KnownObjectTable::Index adIndex)
+   {
+   TR::VMAccessCriticalSection getMethodHandleTableEntryIndex(this);
+   TR::KnownObjectTable::Index result = TR::KnownObjectTable::UNKNOWN;
+   TR::KnownObjectTable *knot = comp->getKnownObjectTable();
+   if (!knot) return result;
+
+   uintptr_t mhTable = 0;
+#if JAVA_SPEC_VERSION == 17
+   uintptr_t typesAndInvokersObj = getReferenceField(knot->getPointer(vhIndex),
+                                                      "typesAndInvokers",
+                                                      "Ljava/lang/invoke/VarHandle$TypesAndInvokers;");
+   if (!typesAndInvokersObj) return result;
+
+   mhTable = getReferenceField(typesAndInvokersObj,
+                                         "methodHandle_table",
+                                         "[Ljava/lang/invoke/MethodHandle;");
+
+#elif JAVA_SPEC_VERSION >= 21
+   mhTable = getReferenceField(knot->getPointer(vhIndex),
+                                          "methodHandleTable",
+                                          "[Ljava/lang/invoke/MethodHandle;");
+#endif
+   if (!mhTable) return result;
+
+   uintptr_t accessDescModeFieldOffset =
+               getInstanceFieldOffset(getObjectClass(knot->getPointer(adIndex)),"mode", "I");
+   int32_t mhEntryIndex = getInt32FieldAt(knot->getPointer(adIndex), accessDescModeFieldOffset);
+   uintptr_t methodHandleObj = getReferenceElement(mhTable, mhEntryIndex);
+
+   // MethodHandle type obtained from the table must match the access descriptor invoker type.
+   uintptr_t methodTypeObj = getReferenceField(methodHandleObj,
```

Comment:
It looks like the method handle table is lazily populated, so `methodHandleObj` could be null here